### PR TITLE
APPTEST-54: Add tests for PatientSetServiceImpl

### PR DESCRIPTION
**Pull request by DrRacket**
Added new test cases in file /api/src/test/java/org/openmrs/api/PatientSetServiceTest.java.
Added annotations for corresponding methods in file /api/src/main/java/org/openmrs/api/impl/PatientSetServiceImpl.java.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@DrRacket this pull request has lots of unnecessary formatting changes. Can you look into that?

**Comment by DrRacket** [comment]
@dkayiwa Cleaned up. Sorry for that.

File: api/src/test/java/org/openmrs/api/PatientSetServiceTest.java [hunk 1]
```diff
@@ -346,4 +348,238 @@ public void getPersonAttributes_shouldReturnPersonAttributesOfTypeLocation() thr
...
+
+	    //this cohort only contains person 8
+	    Cohort cohort = service.getPatientsHavingDateObs(18, ymd.parse("2006-12-31"), ymd.parse("2007-01-02"));
+	    assertEquals(1, cohort.size());
+	    assertTrue(cohort.contains(8));
+	    //this query should only return the obs1 since the obs2 is out of the time range
+	    Map<Integer, List<Obs>> result = service.getObservations(cohort, testConcept,ymd.parse("2006-12-31"),ymd.parse("2007-01-02"));
+	    assertEquals(1, result.size());
+	    assertTrue(result.containsKey(8));
+	    //this assertion fails
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/PatientSetServiceTest.java]: @DrRacket did you look into why this assertion fails?

**Comment by DrRacket** [comment]
@dkayiwa The commented assertion will fail because in the HibernatePatientSetDAO, the method getObservations(Cohort patients, Concept concept, Date fromDate, Date toDate) actually didn't include the fromDate and toDate in the query conditions. The method ignores the date range so for this test case, the method will return 2 observations for patient 8 although 1 observation is  not in the date range.

**Comment by dkayiwa** [comment]
@DrRacket do you think it makes sense to put that explanation in the test? :)

**Comment by DrRacket** [comment]
@dkayiwa I can do that. Is this bug going to be fixed since getObservations(Cohort patients, Concept concept, Date fromDate, Date toDate) has no difference with getObservations(Cohort patients, Concept concept) now.

**Comment by dkayiwa** [comment]
@DrRacket it will be great to have someone look into it. 👍

**Comment by DrRacket** [comment]
@dkayiwa The explanation is added.

**Pull request merged**